### ISSUE TITLE
playwright_include_page not compatible with retry middleware

### ISSUE THREAD

**Issue by Gidgidonihah** [scrapy-plugins/scrapy-playwright#190]
When using [`playwright_include_page`](https://github.com/scrapy-plugins/scrapy-playwright#playwright_include_page) with the RetryMiddleware enabled, scrapy can get stuck, as noted in [notes about leaving unclosed pages](https://github.com/scrapy-plugins/scrapy-playwright#receiving-page-objects-in-callbacks).

If you are including the playwright page in your request, and the request needs to be retried, the page will not be closed. This will lead to hitting the [`PLAYWRIGHT_MAX_PAGES_PER_CONTEXT`](https://github.com/scrapy-plugins/scrapy-playwright#playwright_max_pages_per_context) setting and causing the crawl to get stuck.

I didn't spot a way around this, but I am happy to have one pointed out to me. It seems that this may just be something we have to deal with, so it may make sense to include a custom middleware that subclasses the retry middleware and handles closing the page when present on a retry.

**Comment by elacuesta** [comment]
Sounds reasonable, but I'll need some code to reproduce. Could you provide a [minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)?

**Comment by Gidgidonihah** [comment]
Sorry, I just spotted today that you asked for an MRE. Reasonable.

I tried to throw one together from scratch with no luck. Even going back to the in-development code as it was before I reported this ticket, I'm unable to reproduce, so I'll go ahead and close the ticket.

What I've found is that the callback always gets called, even after the (default) 3 retries. As long as I'm closing the page in the callback (and the errback) my page is getting closed. So at the time, something must have been preventing the callback from being called. This is what I remember seeing, and logically what makes sense, but unfortunately, I don't remember why. I should have included more information originally.

The way I tested just now was by forcing retries with `"RETRY_HTTP_CODES": range(200, 599)`. Perhaps I just wasn't able to duplicate the reasons for the retry that was causing the for me while under development.

Regardless, consider this not a thing unless I (or someone else) posts back and reopens this.
